determined-ai/determined
# feat: Check for model edit permission when registering checkpoint [WEB-808]

## Description

OSS portion of WEB-808 - prep to filter the list of models returned by whether user has permission to edit - may end up getting tied to another ticket re: permission to view?

Changes
- GetModels API adds editable boolean param
- Register Checkpoint modal passes editable=true when getting models
- bun-ify the server's GetModels query and the name-duplication check (allows removing get_models.sql)
- Adds `FilterEditableModelsQuery` to authz, based on experiments' `FilterExperimentsQuery`, but in OSS this only needs a pass-through

## Test Plan

Model registry
- On the Model Registry page, note that models appear and can be sorted by name or by time
- See two models (example: Model 1 and Model 2).  Click on one of them (Model 1) and attempt renaming it to the existing name (Model 2). Error message should appear. Refreshing the page shows us the unchanged model name.

Checkpoint registration
- On a successful experiment, find a checkpoint and click the flag button.
- Open the browser console / network tab.
- Continue as if registering a model. When you open the model selection modal, it should make a request to `/api/v1/models` including `editable=true`
- On the model selection modal, you should see a list of models

## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
